studio/app/optinist/routers/mat.py

```diff
@@ -0,0 +1,76 @@
+from functools import reduce
+from typing import List
+
+import numpy as np
+from fastapi import APIRouter
+from scipy.io import loadmat
+
+from studio.app.common.core.utils.filepath_creater import join_filepath
+from studio.app.dir_path import DIRPATH
+from studio.app.optinist.routers.hdf5 import HDF5Getter
+from studio.app.optinist.schemas.mat import MatNode
+
+router = APIRouter()
+
+
+class MatGetter:
+    @classmethod
+    def data(cls, filepath, dataPath: str = None):
+        data = loadmat(filepath, simplify_cells=True)
+        data = {
+            key: value
+            for key, value in data.items()
+            if not key.startswith("__") and not key.startswith("#")
+        }
+        keys = dataPath.split("/") if dataPath is not None else []
```

Comment:
The way mat files are read may vary from version to version.
In v7.3 file, all fields are read as array.
but In matlab viewer, they are not.

This is the info Group in sample_3_ver7_3.mat.
![Screenshot 2024-01-05 at 13 28 46](https://github.com/arayabrain/barebone-studio/assets/42664619/66fc3faa-d4df-4c62-868c-7a429cec3d34)

But it has the following data structure.
![Screenshot 2023-12-25 at 14 58 23](https://github.com/arayabrain/barebone-studio/assets/42664619/040eb9ff-0e6a-420a-be05-77eccd6676fe)


Could you research further?

Reply:
@ReiHashimoto 
Since with v7.3 file h5py reads all fields as arrays, I use `pymatreader` instead.
It returns the data type as expected and works with both the old (< 7.3) and the new (>= 7.3) matlab version.